blocks/show-hide-group/front-end.js

```diff
@@ -0,0 +1,43 @@
+function docReady(fn) {
+	// see if DOM is already available
+	if (
+		document.readyState === 'complete' ||
+		document.readyState === 'interactive'
+	) {
+		// call on next available tick
+		setTimeout(fn, 1);
+	} else {
+		document.addEventListener('DOMContentLoaded', fn);
+	}
+}
+
+const details = document.querySelectorAll(
+	'.wp-block-happyprime-show-hide-group details'
+);
+
+docReady(function () {
+	const toggleAll = document.querySelectorAll(
+		'.wp-block-happyprime-show-hide-group .toggle-all'
+	);
+	if (toggleAll.length === 1) {
+		toggleAll[0].addEventListener('click', () => {
```

Comment:
Multiple block support:
Not accounting for coding standards 🙂. Additionally, the definition of the `details` constant on line 14 would be removed, and `toggleAll[0]` from this point on would be replaced by `toggle`.
```suggestion
		toggleAll.forEach((toggle) => toggle.addEventListener('click', () => {
			const details = toggle.parentElement.querySelectorAll(
				'.wp-block-happyprime-show-hide-group details'
			);
```